modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ItemActionForm.tsx

```diff
@@ -332,6 +353,36 @@ const ItemActionForm = ({
 								/>
 							</ClayForm.Group>
 
+							<ClayForm.Group>
+								<label htmlFor={permissionKeyFormElementId}>
+									{Liferay.Language.get(
+										'headless-action-key'
+									)}
+
+									<span
+										className="label-icon lfr-portal-tooltip ml-2"
+										title={Liferay.Language.get(
+											'headless-action-key-help'
+										)}
+									>
+										<ClayIcon symbol="question-circle-full" />
+									</span>
+								</label>
+
+								<ClayInput
+									id={permissionKeyFormElementId}
+									onChange={(event) =>
```

Comment:
Not a blocker, but I've seen some `<ClayInput onChange= .... >` for simple input texts like url, permissionkey and also in other screens. This generates a state change on each keystroke which we probably don't need.

Reply:
@bryceosterhaus and I had a discussion on one PR about this, but I can't find it now. refs vs. state is basically performance vs. simplicity and readability. I was defending refs at the time, but nowadays I'm leaning more towards state. In the end, performance loss seems pretty negligible. Maybe if state would wrap a huge component it would become an issue, but I have yet to see this case in real use.